### PR TITLE
Removed first batch from default algorithm

### LINKED ISSUE #4219: Remove DefaultAlgorithm Batch 1

### PR DESCRIPTION
**Pull request by christopherbunn**
Resolves #4219

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Review by jeremyliweishih — APPROVED** [review]
LGTM

File: evalml/automl/automl_algorithm/default_algorithm.py [hunk 1]
```diff
@@ -169,11 +169,11 @@ def __init__(
     def default_max_batches(self):
         """Returns the number of max batches AutoMLSearch should run by default."""
         if self.ensembling:
-            return 4
+            return 3
         elif is_time_series(self.problem_type):
             return 2  # we do not run feature selection for time series
         else:
-            return 3
+            return 2
```
Comment (eccabay) [inline evalml/automl/automl_algorithm/default_algorithm.py:176]: Looks like we can combine these two cases now!

File: docs/source/start.ipynb [hunk 2]
```diff
@@ -1,13 +1,15 @@
 {
  "cells": [
   {
+   "attachments": {},
```
Comment (eccabay) [inline docs/source/start.ipynb:4]: Nit: can you back out all these extra args 😅

File: evalml/tests/automl_tests/test_automl.py [hunk 3]
```diff
@@ -1805,7 +1804,8 @@ def test_pipelines_in_batch_return_nan(
         X_train=X,
         y_train=y,
         problem_type="binary",
-        max_batches=3,
+        max_batches=2,
+        automl_algorithm="iterative",
```
Comment (eccabay) [inline evalml/tests/automl_tests/test_automl.py:1808]: Why was this change required?

File: evalml/tests/automl_tests/test_automl.py [hunk 4]
```diff
@@ -1819,7 +1819,10 @@ def test_pipelines_in_batch_return_nan(
             automl.search()
     assert len(automl.errors) > 0
     for pipeline_name, pipeline_error in automl.errors.items():
-        assert "Label Encoder" in pipeline_error["Parameters"]
+        assert (
+            "Label Encoder" in pipeline_error["Parameters"]
+            or "Mock Classifier" in pipeline_error["Parameters"]
+        )
```
Comment (eccabay) [inline evalml/tests/automl_tests/test_automl.py:1825]: Same question here, why did this have to change?

File: docs/source/start.ipynb (hunk 2 above)
Comment (christopherbunn) [reply docs/source/start.ipynb:4]: These additional cell attachment tags are going to be the end of me 😭. I'm opened [a new issue](https://github.com/alteryx/evalml/issues/4222) to get this added to our linting tools.

File: evalml/tests/automl_tests/test_automl.py (hunk 4 above)
Comment (christopherbunn) [reply evalml/tests/automl_tests/test_automl.py:1825]: Previously, our mock didn't actually work. We now check to see that the mock classifier pipelines appear in the results.

File: evalml/tests/automl_tests/test_automl.py (hunk 3 above)
Comment (christopherbunn) [reply evalml/tests/automl_tests/test_automl.py]: It was set to run the default algorithm even though we had the iterative one mocked out! I updated it so that we're calling the mocks now.
